cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_uniform_dpc.cpp

```diff
@@ -0,0 +1,367 @@
+        auto [inp_array, inp_event] = input();
+        auto [out_array, out_event] = output(height);
+
+        const float_t* inp_ptr = inp_array.get_data();
+        float_t* out_ptr = out_array.get_mutable_data();
+
+        reduction_t reducer(this->get_queue());
+        reducer(inp_ptr,
+                out_ptr,
+                width,
+                height,
+                stride,
+                binary_t{},
+                unary_t{},
+                { inp_event, out_event })
+            .wait_and_throw();
+
+        check_output_rw(out_array);
+    }
+
+    void test_raw_cw_reduce_naive() {
+        using reduction_t = reduction_rm_cw_naive<float_t, binary_t, unary_t>;
+        auto [inp_array, inp_event] = input();
+        auto [out_array, out_event] = output(width);
+
+        const float_t* inp_ptr = inp_array.get_data();
+        float_t* out_ptr = out_array.get_mutable_data();
+
+        reduction_t reducer(this->get_queue());
+        reducer(inp_ptr,
+                out_ptr,
+                width,
+                height,
+                stride,
+                binary_t{},
+                unary_t{},
+                { inp_event, out_event })
+            .wait_and_throw();
+
+        check_output_cw(out_array);
+    }
+
+    void test_raw_cw_reduce_naive_local() {
+        using reduction_t = reduction_rm_cw_naive_local<float_t, binary_t, unary_t>;
+        auto [inp_array, inp_event] = input();
+        auto [out_array, out_event] = output(width);
+
+        const float_t* inp_ptr = inp_array.get_data();
+        float_t* out_ptr = out_array.get_mutable_data();
+
+        reduction_t reducer(this->get_queue());
+        reducer(inp_ptr,
+                out_ptr,
+                width,
+                height,
+                stride,
+                binary_t{},
+                unary_t{},
+                { inp_event, out_event })
+            .wait_and_throw();
+
+        check_output_cw(out_array);
+    }
+
+    void test_raw_cw_reduce_wrapper() {
+        using reduction_t = reduction_rm_cw<float_t, binary_t, unary_t>;
+        auto [inp_array, inp_event] = input();
+        auto [out_array, out_event] = output(width);
+
+        const float_t* inp_ptr = inp_array.get_data();
+        float_t* out_ptr = out_array.get_mutable_data();
+
+        reduction_t reducer(this->get_queue());
+        reducer(inp_ptr,
+                out_ptr,
+                width,
+                height,
+                stride,
+                binary_t{},
+                unary_t{},
+                { inp_event, out_event })
+            .wait_and_throw();
+
+        check_output_cw(out_array);
+    }
+
+    auto get_width() const {
+        return width;
+    }
+
+    auto get_stride() const {
+        return stride;
+    }
+
+    auto get_height() const {
+        return height;
+    }
+
+private:
+    float_t arg;
+    std::int64_t width;
+    std::int64_t stride;
+    std::int64_t height;
```

Comment:
We use the same coding style across tests, so:
```suggestion
    float_t arg_;
    std::int64_t width_;
    std::int64_t stride_;
    std::int64_t height_;
```